Fix: Show “Add” button instead of “Edit” in Prescription section

aakifey opened the pull request:
## Proposed Changes

- Fixes #12727 
- Shows Add button when no active prescriptions.

https://github.com/user-attachments/assets/7b1ebaca-08d8-46f2-afc5-57314296de7f



- Incase the user wants to edit stopped prescriptions(add notes), changes add to edit to avoid confusion. *Removed

https://github.com/user-attachments/assets/1534ab47-929b-406c-8512-073312f50f21



@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [x] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [x] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [x] Request for Peer Reviews
- [x] Completion of QA in Mobile Devices
- [x] Completion of QA in Desktop Devices


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## Summary by CodeRabbit

* **New Features**
  * The medication table's edit button now dynamically switches to an "add" button with a plus icon and label when no active medications are present.

* **Tests**
  * Enhanced test commands to support clicking buttons labeled either "Add" or "Edit" for better interaction coverage.

* **Style**
  * Updated button icons and labels in the medication table for clearer user guidance based on medication status.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

aakifey committed:
fixed add/edit issue

aakifey committed:
Merge branch 'develop' into issues/12727/show-add-button-instead-of-edit

AdityaJ2305 reviewed (changes requested):
- [ ] It shouldn't be dependent of `showStopped` show `add` when no active medicine independent of showStopped state
 - [ ] directly use condition instead of declaring variable 
 - [ ] we can use it once right ? don't need to declare twice one for icon and second for text

aakifey committed:
added changes requested

aakifey committed:
Merge branch 'ohcnetwork:develop' into issues/12727/show-add-button-instead-of-edit

AdityaJ2305 commented on src/components/Medicine/MedicationRequestTable/index.tsx:
```diff
@@ -197,8 +197,17 @@ export default function MedicationRequestTable({ patient, encounter }: Props) {
                       <Link href={`questionnaire/medication_request`}>
-                        <PencilIcon className="mr-2 size-4" />
-                        {t("edit")}
+                        {!activeMedications?.results?.length ? (
```
If `activeMedications?.results` is undefined, .length will throw an error since you’re trying to access .length of undefined

aakifey replied:
it would evaluate as !undefined right, which would return true and avoid throwing an error

AdityaJ2305 replied:
Oh my bad , didn't notice `?` on result do resolve the conservation 👍

Jacobjeevan reviewed (approved):
- Edit the merge checklist (check anything applicable that you have done) @aakifey 

LGTM otherwise.

aakifey committed:
Merge branch 'develop' into issues/12727/show-add-button-instead-of-edit

aakifey committed:
Merge branch 'develop' into issues/12727/show-add-button-instead-of-edit

aakifey committed:
Merge branch 'develop' into issues/12727/show-add-button-instead-of-edit

aakifey committed:
Updated tests

aakifey committed:
Merge branch 'issues/12727/show-add-button-instead-of-edit' of https://github.com/aakifey/care_fe into issues/12727/show-add-button-instead-of-edit

aakifey committed:
Merge branch 'develop' into issues/12727/show-add-button-instead-of-edit

aakifey commented:
Updated tests to accommodate the changes made:

- Added RegExp to verifyAndClickElement

- RegExp helps to deal with the changing of Add and Edit


![Screenshot 2025-07-01 231814](https://github.com/user-attachments/assets/a525126c-7288-4c0f-8d68-9c6900160d9d)

nihal467 commented:
LGTM
